### DISCUSSION TITLE
[Question] How to create a standalone parser for floating-point values?

### DISCUSSION THREAD

**Discussion by willowell** [j-mie6/Parsley#218] [Q&A] [answered]
Hello!

How can I create a standalone parser for floating-point values? e.g.,

```scala
val double: Parsley[Double] = ???
```

I'm porting my Advent of Code solutions over to Scala and building a foundation for this year's Advent of Code. So, I would like to have a simple parser for floating-point values on-hand, just in case.

I see that I could create a parser for floating-point values by creating a lexer (as described here https://github.com/j-mie6/Parsley/wiki/Building-a-Parser-for-Haskell#lexing), but this feels a bit over-kill for my use-case.

Is this the best way to approach this with Parsley or is there a simpler way?

Thank you!

**Comment by j-mie6** [comment, marked as answer]
The `Lexer` is absolutely the simplest way to do this in terms of LOC:

```scala
val lexer = new Lexer(LexicalDesc.plain)
val double: Parsley[Double] = lexer.nonlexeme.numeric.floating.doubleRounded // or just `double` if you want it to ensure it's a valid Java Double
```

Even with the default configuration this is probably good for most needs (`nonlexeme` here to not consume any whitespace).
If you don't want that, you'd just have to implement your own floating point parser, which isn't too hard but is more work than this solution for sure 😛

**Reply by j-mie6** [reply]
(note that the wiki is still a tad out of date for the lexer configuration: working on it!)

**Reply by willowell** [reply]
Hello @j-mie6 !

Thank you so much for replying so quickly and for clarifying!

That's great to know - seems just as well; I've found the integer parses under `lexer.nonlexeme.numeric.integer` to be indispensable so far!
